3_core_core_analysis/5_KEGG_enrichment_of_stable_genes.py

```diff
@@ -144,17 +146,86 @@ def KEGG_enrichment_of_stable_genes(similarity_score_df, gene_list, kegg_df):
 print(pao1_least_stable_enrichment.shape)
 pao1_least_stable_enrichment.sort_values(by="corrected p-value").head()
 
-# TO DO: Remove 'compare' when we decide which input to use
-# Save
-pao1_most_stable_enrichment.to_csv("pao1_most_stable_enrichment_compare.tsv", sep="\t")
-pao1_least_stable_enrichment.to_csv(
-    "pao1_least_stable_enrichment_compare.tsv", sep="\t"
+# ## Plot
+
+# +
+# Plot top most stable
+pao1_most_stable_enrichment_top = pao1_most_stable_enrichment.sort_values(
+    by="corrected p-value"
+).head(10)
+# pval_lst = list(
+# pao1_most_stable_enrichment_top.sort_values(by="corrected p-value", ascending=False)["corrected p-value"])
+
+plt.figure(figsize=(8, 6))
+f = sns.scatterplot(
+    data=pao1_most_stable_enrichment_top,
+    x="odds ratio",
+    y="enriched KEGG pathway",
+    s=300,
+    # size="corrected p-value",
+    hue="corrected p-value",
+    # hue_order=pval_lst,
+    # size_order=pval_lst,
+    # sizes=(100, 10),
+    legend="full",
+)
+plt.legend(
+    title="corrected p-value",
+    title_fontsize=14,
+    fontsize=12,
+    bbox_to_anchor=(1.05, 0.8),
+    loc="upper left",
+    borderaxespad=0,
+)
+
+f.set_title("Enrichment of most stable core genes", fontsize=16)
+plt.xticks(fontsize=14)
+plt.yticks(fontsize=14)
+f.set_xlabel("Odds ratio", fontsize=16)
+f.set_ylabel("")
+
+# +
+# Plot top most stable
+pao1_least_stable_enrichment_top = pao1_least_stable_enrichment.sort_values(
+    by="corrected p-value"
+).head(10)
+plt.figure(figsize=(8, 6))
+g = sns.scatterplot(
+    data=pao1_least_stable_enrichment_top,
+    x="odds ratio",
+    y="enriched KEGG pathway",
+    s=300,
+    # size="corrected p-value",
+    hue="corrected p-value",
+    # hue_order=pval_lst,
+    # size_order=pval_lst,
+    # sizes=(100, 10),
+    legend="full",
 )
+plt.legend(
```

Comment:
Might be good to round these - having this many digits after the decimal doesn't really add much IMO.